airbyte-cdk/java/airbyte-cdk/db-destinations/src/main/java/io/airbyte/cdk/integrations/destination/jdbc/typing_deduping/JdbcSqlGenerator.java

```diff
@@ -62,8 +62,10 @@ protected DataType<?> toDialectType(final AirbyteType type) {
 
   protected DataType<?> toDialectType(final AirbyteProtocolType airbyteProtocolType) {
     return switch (airbyteProtocolType) {
-      case STRING -> SQLDataType.VARCHAR;
-      case NUMBER -> SQLDataType.FLOAT;
+      // Many destinations default to a very short length (e.g. Redshift defaults to 256)
+      case STRING -> SQLDataType.VARCHAR(65535);
+      // We default to precision=39, scale=9 across destinations
```

Comment:
Is this 39 supposed to be 38? Also where does the 9 come from?

Reply:
> 39

.... thanks copilot 😅 

> 9

mostly arbitrary. I think this is what redshift defaults to if you don't specify the parameters, and [bigquery](https://cloud.google.com/bigquery/docs/reference/standard-sql/data-types#decimal_types) uses the same settings.